inst/www/shared/shiny_scss/shiny.bootstrap5.scss

```diff
@@ -38,8 +38,15 @@ $datepicker-disabled-color:         $dropdown-link-disabled-color !default;
 
 $shiny-file-active-shadow:          $input-focus-box-shadow !default;
 
-/* Treat conditional panels and uiOutput as "pass-through" containers */
+
 .shiny-panel-conditional,
 div:where(.shiny-html-output) {
-  display: contents;
+  /* uiOutput()/ conditionalPanel() are "pass-through" containers when they have children. */
+  &:has(> *) {
+    display: contents;
+  }
+  /* And since they display:contents, forward any styles that are applied to them to their children */
+  &.recalculating > * {
+    opacity: 0.3;
+  }
```

Comment:
Should this be in the `&:has(> *)` block?

The comment is a little confusing because "forward any styles" sounds global and automatic but we're manually applying an important style (`.recalculating`) that is broken by `display: contents`.

Reply:
Oh right, that makes sense. I think the intent is clearer when its nested. Also there's the possibility that the browser doesn't support `:has()`, in which case the recalculating opacity would be applied twice.